### PR TITLE
feat!: make outbound aspect explicit

### LINKED ISSUE #29: Make outbound aspect more explicit
__What should we do?__

* Move outbound connector API to the `outbound.api` package
* Rename `ConnectorFunction` to `OutboundConnectorFunction`
* Clearly document the _outbound_ bits as part of the SDK

__Why should we do it?__

To prevent future migration we can already prepare for inbound to join the SDK.

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

Move classes to `outbound` package where reasonable.
Adds `Outbound` aspect to classes where reasonable.
Moves secret and validation API to respective packages as well.

## Related issues

closes #29

### CODE REVIEW

File: core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java [hunk 1]
```diff
@@ -14,13 +14,13 @@
...
 
 import io.camunda.connector.api.annotation.Secret;
-import io.camunda.connector.impl.TestInput;
+import io.camunda.connector.impl.outbound.OutboundTestInput;
 
-public class InputNestedObject {
-  @Secret public final TestInput secretContainer = new TestInput();
-  @Secret public final TestInput nullContainer = null;
-  public final TestInput otherProperty = new TestInput();
+public class OutboundInputNestedObject {
```
Comment (nikku) [inline core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java:22]: Shall we really prefix all these test classes? Why not have them contained as annonymous nested classes in the spec file (and skip prefixing).

File: core/src/test/java/io/camunda/connector/test/outbound/OutboundConnectorFunctionTest.java [hunk 2]
```diff
@@ -41,12 +42,12 @@ public void shouldExecuteConnector() throws Exception {
...
     // given
-    var fn = new ExampleFunction();
+    var fn = new ExampleOutboundFunction();
 
     // when
     var context =
-        ConnectorContextBuilder.create()
-            .variables(new ExampleInput("secrets.FOO"))
+        OutboundConnectorContextBuilder.create()
+            .variables(new ExampleOutboundInput("secrets.FOO"))
```
Comment (nikku) [inline core/src/test/java/io/camunda/connector/test/outbound/OutboundConnectorFunctionTest.java:50]: Why `Outbound` prefix such test utilitiess?

**Review by nikku — COMMENTED** [review]
Looks good to me. Just questioning if we really need to prefix _All_ classes contained within these packages.

I'd personally not prefix every (internal) test utility.

File: core/src/test/java/io/camunda/connector/test/outbound/OutboundConnectorFunctionTest.java (hunk 2 above)
Comment (tmetzke) [reply core/src/test/java/io/camunda/connector/test/outbound/OutboundConnectorFunctionTest.java:50]: Debatable, I agree. The thing is, this is creating a connector context for outbound connectors. People building custom connectors will also use this.

Comment (tmetzke) [reply core/src/test/java/io/camunda/connector/test/outbound/OutboundConnectorFunctionTest.java:50]: Outbound and Inbound context will most probably be a bit different (though sharing some capabilities, for sure).

File: core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java (hunk 1 above)
Comment (tmetzke) [reply core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java:22]: Yep, I like that idea 👍

**Review by igpetrov — APPROVED** [review]

File: core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java (hunk 1 above)
Comment (tmetzke) [reply core/src/test/java/io/camunda/connector/impl/outbound/secrets/OutboundInputNestedObject.java]: See [03b4bac](https://github.com/camunda/connector-sdk/pull/80/commits/03b4bacbfd4d358fb0ec03d8f17c4e953767b286)

**Review by nikku — APPROVED** [review]

**Review by nikku — APPROVED** [review]

**Pull request merged**